addons/sourcemod/scripting/Leader.sp

```diff
@@ -223,12 +291,120 @@ public void OnConfigsExecuted()
 	PrecacheGeneric(FollowVMT, true);
 	PrecacheGeneric(TrailVTF, true);
 	PrecacheGeneric(TrailVMT, true);
+
+	PrecacheModel("models/oylsister/misc/signal_marker.mdl", true);
+	PrecacheModel("models/oylsister/misc/signal_marker.dx80.vtx", true);
+	PrecacheModel("models/oylsister/misc/signal_marker.dx90.vtx", true);
+	PrecacheModel("models/oylsister/misc/signal_marker.sw.vtx", true);
+	PrecacheModel("models/oylsister/misc/signal_marker.vvd", true);
+
+	PrecacheModel("materials/oylsister/signal_marker/material0.vtf", true);
+	PrecacheModel("materials/oylsister/signal_marker/material0.vmt", true);
+
+	AddFileToDownloadsTable("models/oylsister/misc/signal_marker.mdl");
+	AddFileToDownloadsTable("models/oylsister/misc/signal_marker.dx80.vtx");
+	AddFileToDownloadsTable("models/oylsister/misc/signal_marker.dx90.vtx");
+	AddFileToDownloadsTable("models/oylsister/misc/signal_marker.sw.vtx");
+	AddFileToDownloadsTable("models/oylsister/misc/signal_marker.vvd");
+
+	AddFileToDownloadsTable("materials/oylsister/signal_marker/material0.vtf");
+	AddFileToDownloadsTable("materials/oylsister/signal_marker/material0.vmt");
+
+	leaders_max = g_cVMaxLeaders.IntValue;
+}
+
+//----------------------------------------------------------------------------------------------------
+// Purpose:
+//----------------------------------------------------------------------------------------------------
+
+public void OnClientPutInServer(int client)
+{
+	g_fNextSpawn[client] = 0.0;
+	g_bIsClientALeader[client] = false;
+	g_iClientNextVote[client] = 0;
 }
 
+public void OnClientCookiesCached(int client)
+{
+	char sBool[6], sBool2[6], sColor[128], sButton[32];
+
+	GetClientCookie(client, g_hClientLightEnabled, sBool, sizeof(sBool));
+	GetClientCookie(client, g_hClientMarkerEnabled, sBool2, sizeof(sBool2));
+	GetClientCookie(client, g_hClientColor, sColor, sizeof(sColor));
+	GetClientCookie(client, g_hClientButton, sButton, sizeof(sButton));
+
+	g_bHasLightInMarker[client] = true;
+	g_bIsMarkerEnabled[client] = true;
+
+	if(!StrEqual(sBool, ""))
+	{
+		g_bHasLightInMarker[client] = view_as<bool>(StringToInt(sBool));
+	}
+
+	if(!StrEqual(sBool2, ""))
+	{
+		g_bIsMarkerEnabled[client] = view_as<bool>(StringToInt(sBool2));
+	}
+
+	if(StrEqual(sColor, "0"))
```

Comment:
Convert `sColor` to an integer and call the function `GetColorByNum` from this comment:
- https://github.com/srcdslab/sm-plugin-Leader/pull/2/files#r1141955857